radix-engine/src/blueprints/transaction_processor/tx_processor.rs

```diff
@@ -614,17 +614,17 @@ impl TransactionProcessorBlueprint {
 }
 
 struct TransactionProcessor<'blob> {
-    proof_id_mapping: HashMap<ManifestProof, ObjectId>,
-    bucket_id_mapping: HashMap<ManifestBucket, ObjectId>,
+    proof_id_mapping: IndexMap<ManifestProof, ObjectId>,
```

Comment:
Why do we need need `IndexMap` here?

Reply:
Because we iterate over this one at:
https://github.com/radixdlt/radixdlt-scrypto/blob/ca80a43d162461c2bed48f501aa9440d30f6fcad/radix-engine/src/blueprints/transaction_processor/tx_processor.rs#L216

(I was not analyzing whether the particular iteration has potential of causing determinism or not; I simply followed the migration algo described at https://github.com/radixdlt/radixdlt-scrypto/pull/931#issue-1644373644)